usaspending_api/accounts/v2/views/federal_account_program_activities_total.py

```diff
@@ -0,0 +1,86 @@
+from django.db.models import Case, Q, TextField, When, Value, Sum
+from django.db.models.functions import Coalesce
+from rest_framework.response import Response
+from rest_framework.request import Request
+
+from usaspending_api.accounts.v2.views.federal_account_base import FederalAccountBase, PaginationMixin
+from usaspending_api.common.helpers.generic_helper import get_pagination_metadata
+from usaspending_api.financial_activities.models import FinancialAccountsByProgramActivityObjectClass
+
+
+class FederalAccountProgramActivitiesTotal(PaginationMixin, FederalAccountBase):
+    """
+    Retrieve a list of program activity totals for a federal account.
+    """
+
+    endpoint_doc = (
+        "usaspending_api/api_contracts/contracts/v2/federal_accounts/federal_account_totals/program_activities.md"
+    )
+    filters: dict
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.default_sort_column = "obligations"
+        self.sortable_columns = ["obligations", "code", "name", "type"]
+
+    def post(self, request: Request, *args, **kwargs):
+        self.filters = request.data.get("filter", [])
```

Comment:
Need to implement the TinyShield validation for the filters in the request body. We have a couple of patterns for this that you can lean on (such as this example in [agency_base.py](https://github.com/fedspendingtransparency/usaspending-api/blob/master/usaspending_api/agency/v2/views/agency_base.py#L32)).

Additionally, can you make sure to expect a plural "filters" instead of a singular "filter".

Reply:
If you could implement this validation similar to the agency_base.py example that will allow us to carry forward your logic for all federal account endpoints we need to create.